src/libraries/Microsoft.PowerFx.Core/Binding/BinderUtils.cs

```diff
@@ -672,12 +673,63 @@ BinderCheckTypeResult ReportInvalidOperation()
 
         private static BinderCheckTypeResult CheckComparisonArgTypesCore(IErrorContainer errorContainer, TexlNode left, TexlNode right, DType typeLeft, DType typeRight)
         {
+            // Special case for number-backed BuiltIn Option Sets:
```

Comment:
>pecial case for number-backed BuiltIn Option Sets [](http://example.com/codeflow?start=16&length=49)

We should have tests (.txt) for this. Something like

    >>> If(ErrorKind.Numeric = 24, "yes", "no")
    "yes"

    >>> If(13 = ErrorKind.Div0, "yes", "no")
    "yes"

    >>> If(ErrorKind.Div0 < 10, "yes", "no")
    "no"
    ...

Reply:
Nevermind, found those